File: backend/src/main/java/bizworks/backend/dtos/SearchDTO.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package bizworks.backend.dtos;

import java.util.Date;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 *
 * @author PC
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
public class SearchDTO {

  private Date startDate;
  private Date endDate;
  private String leaveType;
  private String employeeName;
  private String status;
}

File: mobile/mobile/lib/models/SearchDTO.dart
class SearchDTO {
  final DateTime? startDate;
  final DateTime? endDate;
  final String? leaveType; // Chuyển thành String
  final String? status;

  SearchDTO({this.startDate, this.endDate, this.leaveType, this.status});

  Map<String, dynamic> toJson() {
    return {
      'startDate': startDate?.toIso8601String(),
      'endDate': endDate?.toIso8601String(),
      'leaveType': leaveType,
      'status': status,
    };
  }
}
